Add RELEASES procedures to repo (#189)
CosmosContracts/juno: RELEASES.md

# Release and Upgrade procedures and communications

> The `#validators-private` channel on discord will be used for all communications from the team. Only active validators should be allowed access, for security reasons.

Reviewer: ```suggestion
The `#validators-private` channel on discord will be used for all communications from the team. Only **active validators** should be allowed access, for security reasons.
```

Reviewer: adding emphasis

> **The core team will endeavour to always make sure there is 48-72 hours' notice of an impending upgrade, unless there is no alternative.**

Reviewer: ```suggestion
**The core team will endeavour to always make sure there is 48-72 hours notice of an impending upgrade, unless there is no alternative.**
```

Most of our validator communications is done on the [Juno Discord](https://discord.gg/Juno). You should join, and change your server name to `nick | validator-name`, then ask a mod for permission to see the private validator channels.

## Release versioning

**If a change crosses a major version number, i.e. `3.x.x -> 4.x.x` then it is definitely consensus-breaking.**

In the past, some releases have been consensus-breaking but only incremented a minor version, if clearly indicated. In future we will look to be clearer. 

**Only patch versions, i.e. `x.x.1 -> x.x.2`, or `3.1.0 -> 3.1.1` are guaranteed to be non-consensus breaking.**

## Scheduled upgrade via governance

> For a software upgrade governance proposal:

Reviewer: ```suggestion
For a SoftwareUpgradeProposal via governance:
```

1. Validators will be told via announcements channel when the prop is live
2. Validators will be told via announcements channel if it passes

> 3. Validators will be told via announcements channel when the upgrade instructions are available, and the upgrade will be co-ordinated in the private validators channel as the target upgrade block nears.

Reviewer: ```suggestion
3. Validators will be told via announcements channel when the upgrade instructions are available, and the upgrade will be coordinated in the private validators channel as the target upgrade block nears.
```

## Emergency upgrade or security patch

If the team needs to upgrade the chain faster than the cadence of governance, then a special procedure applies.

> This procedure minimises the amount that is publicly shared about a potential issue.

Reviewer: ```suggestion
This procedure minimizes the amount that is publicly shared about a potential issue.
```
:trollface:
